src/storage/src/sink/kafka.rs

```diff
@@ -910,16 +916,39 @@ impl KafkaSinkState {
     }
 
     /// Report a SinkStatus::Stalled and then halt with the same message.
-    pub async fn halt_on_err<T>(&self, result: Result<T, impl ToString + Debug>) -> T {
+    pub async fn halt_on_err<T>(&self, result: Result<T, anyhow::Error>) -> T {
         match result {
             Ok(t) => t,
-            Err(msg) => {
-                self.update_status(SinkStatus::Stalled(msg.to_string()))
-                    .await;
+            Err(error) => {
+                let hint: Option<String> =
+                    error
+                        .downcast_ref::<RDKafkaError>()
+                        .and_then(|kafka_error| {
+                            if kafka_error.is_retriable()
+                                && kafka_error.code() == RDKafkaErrorCode::OperationTimedOut
+                            {
+                                Some(
+                                    "If you're running a single Kafka broker, ensure \
+                                    that the following configs are set to 1 on the broker:\n\
+                                    - transaction.state.log.replication.factor\n\
+                                    - transaction.state.log.min.isr\n\
+                                    - offsets.topic.replication.factor"
+                                        .to_string(),
```

Comment:
we typically just dont put newlines in errors, and let `psql` or the ui format them!

Reply:
Ah, makes sense, will remove them. It will be easier as well to not having to deal with line breaks while matching these in tests 😅

Reply:
Updated wording and removed linebreaks.